### PR TITLE
Egress Sflow Support

### PR DESCRIPTION
**Pull request by rajkumar38**
**What I did**
Added egress sflow support.
HLD: https://github.com/sonic-net/SONiC/pull/1268

**Why I did it**
New enhancement added to sflow to support egress sampling at port level.
**How I verified it**
Verify the sonic-vs UT scripts. Compile and loaded the swss debian in HW.
**Details if related**
Related PRs:
https://github.com/sonic-net/sonic-utilities/pull/2790
https://github.com/sonic-net/sonic-buildimage/pull/14630

### CODE REVIEW

**Comment by prsunny** [comment]
@dgsudharsan , would you help review this?

**Comment by dgsudharsan** [comment]
@vadymhlushko-mlnx Can you please help review this?

**Comment by venkatmahalingam** [comment]
@Gokulnath-Raja Please review this HLD.

**Review by vadymhlushko-mlnx — APPROVED** [review]

**Comment by prsunny** [comment]
@Gokulnath-Raja , can you please signoff?

**Review by prsunny — APPROVED** [review]

File: cfgmgr/sflowmgr.cpp [hunk 1]
```diff
@@ -106,13 +110,19 @@ void SflowMgr::sflowUpdatePortInfo(Consumer &consumer)
                 speed_change = true;
             }
 
+            string def_dir = "rx";
+            if (m_sflowPortConfMap[key].dir != def_dir)
+            {
+                m_sflowPortConfMap[key].dir = def_dir;
```
Comment (Gokulnath-Raja) [inline cfgmgr/sflowmgr.cpp:116]: Which means user has configured direction, please updated port_info.local_dir_cfg as true.

Comment (rajkumar38) [reply cfgmgr/sflowmgr.cpp:116]: > Which means user has configured direction, please updated port_info.local_dir_cfg as true.

I will change the check as below. Intention is to update default value "rx" when port_info is initialized for first time as it was updated with empty string at line 94.
```
  if (m_sflowPortConfMap[key].dir != def_dir && !port_info.local_dir_cfg )
 {
     m_sflowPortConfMap[key].dir = def_dir;
 }
```
If user has configured the direction and if port speed is changed, in this flow the existing user set direction must be used.

**Comment by rajkumar38** [comment]
@Gokulnath-Raja 
Pushed the changes addressing your comment. Pls check and revert any further comments.

**Review by Gokulnath-Raja — APPROVED** [review]

**Review by prsunny — APPROVED** [review]

**Pull request merged**